rust-runtime/smithy-client/src/bounds.rs

```diff
@@ -123,7 +123,7 @@ pub trait SmithyRetryPolicy<O, T, E, Retry>:
     /// Forwarding type to `E` for bound inference.
     ///
     /// See module-level docs for details.
-    type E: Error + ProvideErrorKind;
+    type E: Error;
```

Comment:
For my understanding, was this change needed so that the IMDS client could re-use the smithy-client retry code?